ast/functions.go

```diff
@@ -382,3 +402,221 @@ func (n *AggregateFuncExpr) Accept(v Visitor) (Node, bool) {
 	}
 	return v.Leave(n)
 }
+
+// Update is used for update aggregate context.
+func (n *AggregateFuncExpr) Update() error {
+	name := strings.ToLower(n.F)
+	switch name {
+	case AggFuncCount:
+		return n.updateCount()
+	case AggFuncFirstRow:
+		return n.updateFirstRow()
+	}
+	return nil
+}
+
+// GetContext get aggregate evaluation context for the current group.
+// If it is nil, add a new context into ContextPerGroup.
+func (n *AggregateFuncExpr) GetContext() *AggEvaluateContext {
+	if n.ContextPerGroup == nil {
+		n.ContextPerGroup = make(map[string](*AggEvaluateContext))
+	}
+	if _, ok := n.ContextPerGroup[n.CurrentGroup]; !ok {
+		c := &AggEvaluateContext{}
+		if n.Distinct {
+			c.Distinct = CreateAggregateDistinct(n.F, n.Distinct)
```

Comment:
CreateAggregateDistinct function has inner checked Distinct value, so seems needless for  n.Distinct  check.